components/formats-api/src/loci/formats/FormatTools.java

```diff
@@ -1437,7 +1437,57 @@ public static Time getTime(Double value, String unit) {
     }
     return new Time(value, UNITS.SECOND);
   }
-  
+
+  /**
+   * Formats the input value for the position into a length.
+   *
+   * @param value  the value of the position
+   * @param unit   the unit of the position. If {@code null} or invalid,
+   *               default to reference frame.
+   *
+   * @return       the physical size formatted as a {@link Length}
+   */
+  public static Length getPosition(Double value, String unit) {
+    if (value == null || value == Double.POSITIVE_INFINITY ||
+        value == -Double.POSITIVE_INFINITY) {
+      LOGGER.debug("Expected float value for position; got {}", value);
+      return null;
+    }
+
+    if (unit == null) return new Length(value, UNITS.REFERENCEFRAME);
+
+    try {
+      return UnitsLength.create(value, UnitsLength.fromString(unit));
+    } catch (EnumerationException e) {
+      LOGGER.debug(e.getMessage());
+      return new Length(value, UNITS.REFERENCEFRAME);
+    }
+  }
+
+  /**
+   * Formats the input value for the position into a length.
+   *
+   * @param value  the value of the position
+   *
+   * @return       the physical size formatted as a {@link Length}
+   */
+  public static Length getPosition(Double value) {
+   return getPosition(value, UNITS.REFERENCEFRAME);
+  }
+
+  /**
+   * Formats the input value for the position into a length of the given unit.
+   *
+   * @param value  the value of the position
+   * @param unit   the unit of the position. If {@code null} or invalid,
+   *               default to reference frame.
+   *
+   * @return       the position formatted as a {@link Length}
+   */
+  public static Length getPosition(Double value, Unit<Length> unit) {
+      return getPosition(value, unit.getSymbol());
```

Comment:
I might need a units refresher here since `UnitsLength.fromString(String)` will return `UnitsLength` and I am looking for a way to come back to `Unit<Length>`. Or should we support both signatures? In many ways, this comes back to our need to document the recommendations for using [the unit system](https://trello.com/c/FvyQj7OP/144-document-unit-changes)